src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -159,6 +159,81 @@ public void MulticastInterface_Set_InvalidIndex_Throws()
             }
         }
 
+        [OuterLoop] // TODO: Issue #11345
+        [Fact]
+        public async Task MulticastInterface_Set_IPv6_AnyInterface_Succeeds()
+        {
+            if (PlatformDetection.IsFedora || PlatformDetection.IsRedHatFamily7 || PlatformDetection.IsOSX)
+            {
+                return; // [ActiveIssue(24008)]
+            }
+
+            // On all platforms, index 0 means "any interface"
+            await MulticastInterface_Set_IPv6_Helper(0);
+        }
+
+        [OuterLoop] // TODO: Issue #11345
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [ActiveIssue(21327, TargetFrameworkMonikers.Uap)] // UWP Apps are forbidden to send network traffic to the local Computer.
+        public async void MulticastInterface_Set_IPv6_Loopback_Succeeds()
+        {
+            // On Windows, we can apparently assume interface 1 is "loopback." On other platforms, this is not a
+            // valid assumption. We could maybe use NetworkInterface.LoopbackInterfaceIndex to get the index, but
+            // this would introduce a dependency on System.Net.NetworkInformation, which depends on System.Net.Sockets,
+            // which is what we're testing here....  So for now, we'll just assume "loopback == 1" and run this on
+            // Windows only.
+            await MulticastInterface_Set_IPv6_Helper(1);
+        }
+
+        private async Task MulticastInterface_Set_IPv6_Helper(int interfaceIndex)
+        {
+            IPAddress multicastAddress = IPAddress.Parse("ff11::1:1");
+            string message = "hello";
+            int port;
+
+            using (Socket receiveSocket = CreateBoundUdpIPv6Socket(out port),
+                          sendSocket = new Socket(AddressFamily.InterNetworkV6, SocketType.Dgram, ProtocolType.Udp))
+            {
+                receiveSocket.ReceiveTimeout = 1000;
+                receiveSocket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.AddMembership, new IPv6MulticastOption(multicastAddress, interfaceIndex));
+
+                // https://github.com/Microsoft/BashOnWindows/issues/990
```

Comment:
Looks like this was supposed to be fixed a while ago.  Can this check be removed?